### ISSUE TITLE
get_loss函数中的问题

### ISSUE THREAD

**Issue by huangyangke** [MUGE-2021/image-retrieval-baseline#12]
我知道下面代码块的作用是聚合不同gpu的特征，然后进行比对损失计算，但是为什么不直接使用gathered_image_features和gathered_text_features呢？
`        # We gather tensors from all gpus to get more negatives to contrast with.
        gathered_image_features = [
            torch.zeros_like(image_features) for _ in range(world_size)
        ]
        gathered_text_features = [
            torch.zeros_like(text_features) for _ in range(world_size)
        ]
        dist.all_gather(gathered_image_features, image_features)
        dist.all_gather(gathered_text_features, text_features)

        all_image_features = torch.cat(
            [image_features]
            + gathered_image_features[:rank]
            + gathered_image_features[rank + 1 :]
        )
        all_text_features = torch.cat(
            [text_features]
            + gathered_text_features[:rank]
            + gathered_text_features[rank + 1 :]
        )`

**Comment by zhzxlcc** [comment]
同样发现 这个gather 和clip的方式不一样，  
all_image_features = torch.cat(gathered_image_features, dim=0)
all_text_features = torch.cat(gathered_text_features, dim=0)
楼主有get是什么原因嘛~
